c++/src/kj/mutex.c++

```diff
@@ -870,7 +870,12 @@ void Once::reset() {
     } \
   }
 
-Mutex::Mutex(): mutex(PTHREAD_RWLOCK_INITIALIZER) {}
+Mutex::Mutex(): mutex(PTHREAD_RWLOCK_INITIALIZER) {
+#if defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070
+  // see https://github.com/capnproto/capnproto/issues/1715
```

Comment:
Sorry to nitpick comments, but this thread is kind of long and meandering and not specific to this issue. Can we just say:

```
  // In older versions of MacOS, mutexes initialized statically cannot be destroyed, so we must
  // call the init function.
```

Reply:
np, here it is. If you wish I may rebase everything into one commit.